### ISSUE TITLE
Consider excluding Duration's associated constants from identity_op lint

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#3866]
This came up in the tracking issue for [`std::time::Duration::SECOND`](https://doc.rust-lang.org/nightly/std/time/struct.Duration.html#associatedconstant.SECOND) and friends: https://github.com/rust-lang/rust/issues/57391.

The following code using `1 * SECOND` arguably reads more clearly in English than `thread::sleep(SECOND)`.

```rust
#![feature(duration_constants)]

use std::thread;
use std::time::Duration;

fn main() {
    thread::sleep(1 * Duration::SECOND);
}
```

```console
warning: the operation is ineffective. Consider reducing it to `Duration::SECOND`
 --> src/main.rs:7:19
  |
7 |     thread::sleep(1 * Duration::SECOND);
  |                   ^^^^^^^^^^^^^^^^^^^^
  |
  = note: #[warn(clippy::identity_op)] on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#identity_op
```

The same pattern is common in Go which uses a similar mechanism for time constants. From a brief search, here are some articles that use `1 * time.Second` or `1*time.Second` in Go:

- https://go.googlesource.com/proposal/+/master/design/12914-monotonic.md
- https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/
- https://dzone.com/articles/practical-golang-getting-started-with-nats-and-rel
- https://divan.dev/posts/go_concurrency_visualize/
- https://dev.to/dizdarevic/golang-the-next-language-to-learn-for-developers-pc5
- http://devs.cloudimmunity.com/gotchas-and-common-mistakes-in-go-golang/
- https://medium.com/code-zen/concurrency-in-go-5fcba11acb0f

This may indicate that this pattern does not fall under the "obscures what’s going on" description of the identity_op lint.

<kbd>*clippy 0.0.212 (5d78250 2019-03-07)*</kbd>

**Comment by oli-obk** [comment]
I agree, we should maybe just whitelist multiplying with constants, since no matter what the constant is, the user is probably doing that multiplication out of a reason.

**Comment by Alexendoo** [comment]
No longer linted: #8204
